### PR TITLE
.*/: handle etcd retryable error when using etcd client's `Get with revision`

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
complete https://github.com/pingcap/dm/issues/517

### What is changed and how it works?
1. Reset get error in watcher like we did in https://github.com/pingcap/dm/pull/499.
2. Get relative etcd info in one etcd transaction to avoid using get with rev.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: dm/worker/server.go [hunk 1]
```diff
@@ -361,11 +365,12 @@ func (s *Server) operateSourceBound(bound ha.SourceBound) error {
 	if bound.IsDeleted {
 		return s.stopWorker(bound.Source)
 	}
-	sourceCfg, _, err := ha.GetSourceCfg(s.etcdClient, bound.Source, bound.Revision)
+	scm, _, err := ha.GetSourceCfg(s.etcdClient, bound.Source, bound.Revision)
 	if err != nil {
 		// TODO: need retry
 		return err
 	}
+	sourceCfg := scm[bound.Source]
```
Comment (csuzhangxc) [inline dm/worker/server.go]: Do we need to check empty sourceCfg here?

File: pkg/ha/stage.go [hunk 2]
```diff
@@ -263,6 +289,32 @@ func subTaskStageFromKey(key string) (Stage, error) {
...
+func subTaskStageFromResp(source, task string, resp *clientv3.GetResponse) (map[string]map[string]Stage, error) {
+	stages := make(map[string]map[string]Stage)
+	if source != "" {
+		stages[source] = make(map[string]Stage) // avoid stages[source] is nil
+	}
+
+	if resp.Count == 0 {
+		return stages, nil
+	} else if source != "" && task != "" && resp.Count > 1 {
+		return stages, fmt.Errorf("too many stage (%d) exist for subtask {sourceID: %s, task name: %s}", resp.Count, source, task)
```
Comment (WangXiangUSTC) [inline pkg/ha/stage.go]: how about add a TODO to use terror?

**Pull request merged**